File: DataBaseRoomApp/jroomapp/src/main/java/com/woowrale/jroomapp/ui/base/BaseActivity.java
package com.woowrale.jroomapp.ui.base;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.woowrale.jroomapp.MainApplication;
import com.woowrale.jroomapp.di.components.MainComponent;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getDaggerMainComponent().inject(this);
    }

    protected MainComponent getDaggerMainComponent() {
        return ((MainApplication) getApplication()).getMainComponent();
    }

    protected abstract void initDagger();

}

File: DataBaseRoomApp/kroomapp/src/main/java/com/woowrale/kroomapp/ui/base/BaseActivity.kt
package com.woowrale.kroomapp.ui.base

import android.app.Application
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.woowrale.kroomapp.MainApplication
import com.woowrale.kroomapp.di.components.MainComponent

abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        getDaggerMainComponent().inject(this)
    }

    protected fun getDaggerMainComponent(): MainComponent {
        val mainApplication: MainApplication = application as MainApplication
        return mainApplication.mainComponent
    }

    protected abstract fun initDagger()
}